packages/jellyfish-api-core/__tests__/category/account/listBurnHistory.test.ts

```diff
@@ -0,0 +1,222 @@
+    expect(history[2].amounts[0]).toStrictEqual('1.00000000@DFI')
+  })
+
+  it('sixth transaction should be send to burn address with accounttoutxos', async () => {
+    const history = await client.account.listBurnHistory()
+    expect(history[1].owner).toStrictEqual(burnAddress)
+    expect(history[1].type).toStrictEqual('None')
+    expect(history[1].amounts.length).toStrictEqual(1)
+    expect(history[1].amounts[0]).toStrictEqual('2.00000000@DFI')
+  })
+
+  it('last transaction should be send utxo to burn address', async () => {
+    const history = await client.account.listBurnHistory()
+    expect(history[0].owner).toStrictEqual(burnAddress)
+    expect(history[0].type).toStrictEqual('None')
+    expect(history[0].amounts.length).toStrictEqual(1)
+    expect(history[0].amounts[0]).toStrictEqual('10.00000000@DFI')
+  })
+
+  it('should listBurnHistory with filter on txtype None', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.NONE
+    })
+    expect(history.length).toStrictEqual(2)
+    expect(history.every(({ type }) => type === 'None')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype UtxosToAccount', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.UTXOS_TO_ACCOUNT
+    })
+    expect(history.length).toStrictEqual(1)
+    expect(history[0].type).toStrictEqual('UtxosToAccount')
+  })
+
+  it('should listBurnHistory with filter on txtype AccountToAccount', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.ACCOUNT_TO_ACCOUNT
+    })
+    expect(history.length).toStrictEqual(2)
+    expect(history.every(({ type }) => type === 'AccountToAccount')).toBeTruthy()
+  })
+
+  it('should listBurnHistory with filter on txtype CreateMasternode', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.CREATE_MASTERNODE
+    })
+    expect(history.length).toStrictEqual(1)
+    expect(history[0].type).toStrictEqual('CreateMasternode')
+  })
+
+  it('should listBurnHistory with filter on txtype CreateToken', async () => {
+    const history = await client.account.listBurnHistory({
+      txtype: DfTxType.CREATE_TOKEN
+    })
+    expect(history.length).toStrictEqual(1)
+    expect(history[0].type).toStrictEqual('CreateToken')
+  })
+
+  it('should listBurnHistory with filter on token DFI', async () => {
+    const history = await client.account.listBurnHistory({
+      token: 'DFI'
+    })
+    expect(history.every(({ amounts }) => amounts[0].includes('DFI'))).toBeTruthy()
+  })
+
+  it('should listBurnHistory with maxBlockHeight 110', async () => {
+    const history = await client.account.listBurnHistory()
+    const historyWithMaxBlockHeight = await client.account.listBurnHistory({
+      maxBlockHeight: 110
+    })
+    expect(Math.max(...history.map(el => el.blockHeight))).toStrictEqual(112)
```

Comment:
I wonder where is this 112 came from.
`waitForWalletCoinbaseMaturity` - 101
setupHistory, `generate(1)` x8
`createToken` has `generate(1)` x2

Reply:
It is not clear this way. I reverted back to a more logical test